core/src/main/scala/kafka/admin/DeleteRecordsCommand.scala

```diff
@@ -61,26 +64,32 @@ object DeleteRecordsCommand {
     if (duplicatePartitions.nonEmpty)
       throw new AdminCommandFailedException("Offset json file contains duplicate topic partitions: %s".format(duplicatePartitions.mkString(",")))
 
+    val recordsToDelete: util.Map[TopicPartition, RecordsToDelete] = new util.HashMap[TopicPartition, RecordsToDelete]
```

Comment:
nit: this foreach thing could be simplified in my opinion:
```
val recordsToDelete = offsetSeq.map(offset => offset._1 -> RecordsToDelete.beforeOffset(offset._2)).toMap.asJava
```

Reply:
Thanks! You are right. We can even skip the direct `var` definition and embed the calculation into the following statement; i.e.

    val deleteRecordsResult = adminClient.deleteRecords(offsetSeq.map(offset => (offset._1, RecordsToDelete.beforeOffset(offset._2))).toMap.asJava)

But I'll leave it as you suggested to make it more readable.
I'll push a commit shortly.